Asterix/optics/phase_amplitude_functions.py

```diff
@@ -1,7 +1,9 @@
+# pylint: disable=invalid-name
+# pylint: disable=trailing-whitespace
+
 import numpy as np
-import Asterix.propagation_functions as prop
-import Asterix.processing_functions as proc
-import Asterix.save_and_read as useful
+from .propagation_functions import mft
```

Comment:
Found one more: you have one relative import left over, this should be an absolute import.

For some resources on relative vs absolute imports (there's many more to be found through googling):
https://stackoverflow.com/a/16748366/10112569

In essence, absolute imports are to be preferred because:
- They future-proof your modules against future rearrangements, needing to change less in that case (and hence breaking less).
- You can use said modules in detached structures like notebooks, which break easily or don't work at all with relative internal imports because they can lead to circular imports.
- It makes it immediately clear where something is coming from.

The only advantage of relative imports that the lines are a little shorter, but Asterix is not a very complex package so it doesn't give you much.

Btw you could make all imports within a package relative imports by just using enough dots (e.g., `from ..utils import xyz`) but please don't make me regret I told you this ^^ This is just to show that the argument of modules living in the same folder is not a strong one in order to use relative imports.